cpp/frontend_pyreadline.cc

```diff
@@ -297,4 +295,27 @@ Str* readline(Str* prompt) {
   return nullptr;
 }
 
+static Str* readline_no_tty(Str* prompt) {
+  mylib::Writer* w = mylib::Stderr();
+  w->write(prompt);
+  w->flush();
+
+  mylib::LineReader* r = mylib::Stdin();
+  Str* s = r->readline();
+  if (s != kEmptyString) {
+    return s;
+  }
+
+  return nullptr;
+}
+
+// See the following for some loose documentation on the approach here:
+// https://tiswww.case.edu/php/chet/readline/readline.html#Alternate-Interface-Example
```

Comment:
Also this comment with bash should now be moved to `readline_tty()`

And then add the comment pointing to CPython for the stdin/stdout conditional can go here